test/syscalls/linux/dev.cc

```diff
@@ -146,6 +158,24 @@ TEST(DevTest, WriteDevFull) {
   EXPECT_THAT(WriteFd(fd.get(), "a", 1), SyscallFailsWithErrno(ENOSPC));
 }
 
+TEST(DevTest, ReadWriteDevKmsg) {
+  const FileDescriptor fd =
+      ASSERT_NO_ERRNO_AND_VALUE(Open("/dev/kmsg", O_WRONLY));
+  constexpr int overMaxSize = 2048;
+  std::vector<char> overMaxBuf(overMaxSize, 1);
+  EXPECT_THAT(WriteFd(fd.get(), overMaxBuf.data(), overMaxSize), SyscallFailsWithErrno(EINVAL));
+  constexpr int bufferSize = 512;
+  std::vector<char> buf(bufferSize, 1);
```

Comment:
Consider using random data as the write payload:

[RandomizeBuffer](https://cs.opensource.google/gvisor/gvisor/+/master:test/util/test_util.h;l=699?q=randomizebuffer&ss=gvisor%2Fgvisor)

```
std::vector<char> buf(bufferSize);
RandomizeBuffer(buf.data(), buf.size());
```